audiobook_generator/core/audiobook_generator.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import os
+
+from audiobook_generator.book_parsers.base_book_parser import get_book_parser
+from audiobook_generator.config.general_config import GeneralConfig
+from audiobook_generator.core.audio_tags import AudioTags
+from audiobook_generator.tts_providers.base_tts_provider import get_tts_provider
+
+logger = logging.getLogger(__name__)
+
+
+def confirm_conversion(price):
+    print(f"Estimate book voiceover would cost you roughly: ${price:.2f}")
+    print("Do you want to continue? (y/n)")
+    answer = input()
+    if answer.lower() != "y":
+        print("Aborted.")
+        exit(0)
+
+
+def get_total_chars(chapters):
+    total_characters = 0
+    for title, text in chapters:
+        total_characters += len(text)
+    return total_characters
+
+
+class AudiobookGenerator:
+    def __init__(self, config: GeneralConfig):
+        self.config = config
+
+    def __str__(self) -> str:
+        return f"{self.config}"
+
+    def run(self):
+        book_parser = get_book_parser(self.config)
+        tts_provider = get_tts_provider(self.config)
+
+        os.makedirs(self.config.output_folder, exist_ok=True)
+        chapters = book_parser.get_chapters(tts_provider.get_break_string())
+        # Filter out empty or very short chapters
+        chapters = [(title, text) for title, text in chapters if text.strip()]
+
+        logger.info(f"Chapters count: {len(chapters)}.")
+
+        # Check chapter start and end args
+        if self.config.chapter_start < 1 or self.config.chapter_start > len(chapters):
+            raise ValueError(
+                f"Chapter start index {self.config.chapter_start} is out of range. Check your input."
+            )
+        if self.config.chapter_end < -1 or self.config.chapter_end > len(chapters):
+            raise ValueError(
+                f"Chapter end index {self.config.chapter_end} is out of range. Check your input."
+            )
+        if self.config.chapter_end == -1:
+            self.config.chapter_end = len(chapters)
+        if self.config.chapter_start > self.config.chapter_end:
+            raise ValueError(
+                f"Chapter start index {self.config.chapter_start} is larger than chapter end index {self.config.chapter_end}. Check your input."
+            )
+
+        logger.info(f"Converting chapters from {self.config.chapter_start} to {self.config.chapter_end}.")
+
+        # Initialize total_characters to 0
+        total_characters = get_total_chars(chapters)
```

Comment:
Not sure if we just count the selected chapters here or the whole book as now.

Reply:
Whole book! all chapters! You need to know the full price of the entire book before you convert it, whats the point of showing the cost of only a single chapter?

Reply:
Makes sense.